app/notification_push.go

```diff
@@ -578,6 +578,21 @@ func (a *App) BuildPushNotificationMessage(contentsConfig string, post *model.Po
 }
 
 func (a *App) SendTestPushNotification(deviceID string) string {
+	canSendPushNotifications := false
+	if *a.Config().EmailSettings.SendPushNotifications {
+		pushServer := *a.Config().EmailSettings.PushNotificationServer
+		if license := a.Srv().License(); pushServer == model.MHPNS && (license == nil || !*license.Features.MHPNS) {
+			mlog.Warn("Push notifications are disabled. Go to System Console > Notifications > Mobile Push to enable them.")
```

Comment:
Nit: Is this a reasonable warning message? Shouldn't the call to action be to upgrade the license or to change the push server? Right now it reads like you just need to enable push notifications, but in reality they're already enabled, right?

Reply:
Agree. This is slightly confusing. We should say to either point to a different push server or upgrade license.

Reply:
This piece of code is taken verbatim from  https://github.com/mattermost/mattermost-server/blob/9d37a6cc3d916e90ca8dd6fd4d61881227707437/app/notification.go#L405

Should the error change also there?

Also, I see some other places where we use slightly different checks (quick search I find it in `app/channels.go line 2907` and `app/expirynotify.go line 20`. Should we make a util function like `func (a *App) canSendPushNotifications() boolean` to unify everything?

Reply:
Great find. Yes, let's make a util method. Thanks Daniel.